src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDAOImpl.java

```diff
@@ -360,4 +366,14 @@ private <T> T execute(String sql, SQLExecution<T> thunk) {
             throw new ShadowManagerDataException(e);
         }
     }
+
+    private <T> T executeWriteOperation(String sql, SQLExecution<T> thunk) {
+        try {
+            return database.getDbWriteThreadPool().submit(() ->
+                    execute(sql, thunk)).get(10, TimeUnit.SECONDS);
```

Comment:
thinking about edge case, is it possible for the execution to take longer than 10 seconds?  not sure what the default execution timeout is but we may want to explicitly set `setQueryTimeout` and then in theory we could just `get()` here

Reply:
Added `setQueryTimeout` on statement but still keeping `get(timeout, seconds)` as we don't know when the thread actually starts executing the query and we can't wait indefinitely.